core/arch/arm/plat-zynqmp/main.c

```diff
@@ -58,6 +58,10 @@ register_phys_mem_pgdir(MEM_AREA_IO_SEC,
 			ROUNDDOWN(GIC_BASE + GICD_OFFSET, CORE_MMU_PGDIR_SIZE),
 			CORE_MMU_PGDIR_SIZE);
 
+register_phys_mem_pgdir(MEM_AREA_IO_SEC,
```

Comment:
Shouldn't this be in CSU driver instead?

Eg. CSU gets mapped only when included in the build.

Reply:
right but the CSU has many subsystems, so I made the choice to initialize the memory here and allow those subsystems to become drivers.